Sources/SpeziHealthKit/Queries/HealthKitQuery.swift

```diff
@@ -0,0 +1,515 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import SwiftUI
+import HealthKit
+import SpeziFoundation
+import Spezi
+
+
+/// The time range for which data should be fetched from the health store.
+public enum HealthKitQueryTimeRange: Hashable, Sendable {
```

Comment:
@PSchmiedmayer i ended up adding probably slightly too many cases here. what's your take as to which time ranges we should want to offer?